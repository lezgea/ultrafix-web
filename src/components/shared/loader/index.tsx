import Image from 'next/image';
import React from 'react';


export const Loader: React.FC = () => {
    return (
        <div className="loader-page-wrapper w-screen min-h-screen bg-white/3">
            <div className="loader-wrapper">
                <div className="overflow-box"></div>
                <Image src="/svg/race.svg" alt="Logo" className="loader" width={80} height={50} priority />
            </div>
        </div>

    )
}