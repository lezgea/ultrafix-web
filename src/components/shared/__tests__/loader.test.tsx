import React from 'react';
import { render } from '@testing-library/react';
import { Loader } from '../loader';

describe('Loader Component', () => {
    it('renders without errors', () => {
        const { asFragment } = render(<Loader />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('displays the loader image', () => {
        const { getByAltText } = render(<Loader />);
        expect(getByAltText('Logo')).toBeInTheDocument();
    });
});
