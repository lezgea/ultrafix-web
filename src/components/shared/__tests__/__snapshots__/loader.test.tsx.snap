// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders without errors 1`] = `
<DocumentFragment>
  <div
    class="loader-page-wrapper w-screen min-h-screen bg-white/3"
  >
    <div
      class="loader-wrapper"
    >
      <div
        class="overflow-box"
      />
      <img
        alt="Logo"
        class="loader"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="50"
        src="/svg/race.svg"
        style="color: transparent;"
        width="80"
      />
    </div>
  </div>
</DocumentFragment>
`;
